android-sdk/src/main/java/com/mobilecoin/lib/MobileCoinClient.java

```diff
@@ -221,11 +222,17 @@ public AccountSnapshot getAccountSnapshot(UnsignedLong blockIndex) throws Networ
         TxOutStore txOutStore = getTxOutStore();
         UnsignedLong storeIndex = txOutStore.getCurrentBlockIndex();
         if (storeIndex.compareTo(blockIndex) < 0) {
-            txOutStore.refresh(
-                    viewClient,
-                    ledgerClient,
-                    fogBlockClient
-            );
+            try {
+                txOutStore.refresh(
+                        viewClient,
+                        ledgerClient,
+                        fogBlockClient
+                );
+            } catch(FogSyncException e) {
+                if(blockIndex.compareTo(e.getFogBlockIndex()) >= 0) {
+                    throw new InvalidFogResponse("Cannot create up-to-date snapshot until Fog sync finishes. Try again later.", e);
```

Comment:
Since it's a fog sync error, I'd rather throw `FogSyncException` instead of `InvalidFogResponse`?

Reply:
OK, I guess we can pass these all the way up.